packages/functions/package.json

```diff
@@ -14,17 +17,27 @@
     "url": "https://github.com/vercel/vercel/issues"
   },
   "devDependencies": {
+    "@aws-sdk/client-s3": "3.590.0",
+    "@aws-sdk/credential-provider-web-identity": "3.587.0",
+    "@smithy/types": "3.0.0",
+    "typescript": "5.4.5",
     "vitest": "1.3.1"
   },
+  "peerDependencies": {
+    "@aws-sdk/credential-provider-web-identity": "*"
```

Comment:
<!-- Questions comments seek clarification.  -->
**question (soft-blocking):** ‏can you explain why we need this and why it needs to be any version (`*`)?

It it in service of the dynamic import? https://github.com/vercel/vercel/pull/11701/files#diff-25c8d9ab60649f508457d1f2005f8cbab25ea20edeced6b857e6a23bdc5c9b69R54

Reply:
@EndangeredMassa Yes, it is in service of the dynamic import.

The AWS SDK imports the same package using `*`: https://github.com/aws/aws-sdk-js-v3/blob/main/packages/credential-provider-node/package.json#L36

If they change this in the future we may need to consider changing it ourselves, but this seems like the safest, future proof path for now.